extensions/GitExtension/FileExplorerGitIntegration/Models/RepositoryWrapper.cs

```diff
@@ -81,13 +111,24 @@ public RepositoryWrapper(string rootFolder)
 
     private CommitLogCache GetCommitLogCache()
     {
-        // Fast path: if we have an up-to-date commit log, return that
+        var result = GitExecute.ExecuteGitCommand(_gitDetect.GitConfiguration.ReadInstallPath(), _workingDirectory, "rev-parse HEAD");
+        if (result.Status != ProviderOperationStatus.Success)
+        {
+            throw result.Ex ?? new InvalidOperationException(result.ProcessExitCode?.ToString(CultureInfo.InvariantCulture) ?? "Unknown error while obtaining HEAD commit");

```

Comment:
Can we provide any information to the user instead of "Unknown error while obtaining HEAD commit"?  Maybe a call to action.  Can they check logs somewhere else?